src/Data/Binary/StringRef.hs

```diff
@@ -47,10 +49,28 @@ instance (StringReferencingBinary a, StringReferencingBinary b, StringReferencin
     ls_get strs                 = liftM5 (,,,,) (ls_get strs) (ls_get strs) (ls_get strs) (ls_get strs) (ls_get strs)
 
 
+newtype CompactNum a = CompactNum { fromCompactNum :: a }
+
+instance (Integral a, Num a, Binary a) => StringReferencingBinary (CompactNum a) where
+    ls_put _ (CompactNum i)
+          | 0 <= i && i < 255 = putWord8 (fromIntegral i)
+          | otherwise         = putWord8 255 >> put i
+    ls_get _ = fmap CompactNum $ getWord8 >>= \case
+        i | 0 <= i && i < 255 -> return (fromIntegral i)
+          | otherwise         -> get
```

Comment:
Did you consider LEB1238-encoding here? Same until 128, worse for `[128…255]`, but then again better.

Reply:
Never heard of it. Not sure if worth the extra complexity though. I expect that in practice most numbers in arbtt logs are quite small, but then I have no idea what other people's environments look like. Do you think it's worth it? Shall I look into it?

Reply:
https://hackage.haskell.org/package/leb128-cereal is a package that I wrote for that a while ago, but we are using a different parsing library here (not `cereal`). Yeah, let’s not bother.